src/compiler/checker.ts

```diff
@@ -14040,8 +14073,29 @@ namespace ts {
                     const declaration = getDeclarationOfAliasSymbol(exportEqualsSymbol) || exportEqualsSymbol.valueDeclaration;
                     error(declaration, Diagnostics.An_export_assignment_cannot_be_used_in_a_module_with_other_exported_elements);
                 }
+                // Checks for export * conflicts
+                const exports = getExportsOfModule(moduleSymbol);
+                for (const id in exports) {
+                    if (id === "__export") {
+                        continue;
+                    }
+                    const exportedSymbol = exports[id];
+                     // 15.2.1.1 It is a Syntax Error if the ExportedNames of ModuleItemList contains any duplicate entries. (TS Exceptions: namespaces, function overloads, enums, and interfaces)

```

Comment:
Based on the way we do should ES6 exports for namespaces and enums, I believe so (the others are types and are fine). Namespaces and enums should end up roughly like this in ES6 forms:
```js
var Namespace_1;
(function(Namespace_1){
 // ...
})(Namespace_1 = Namespace_1 || {});
var Namespace_1;
(function(Namespace_1){
 // ...
})(Namespace_1 = Namespace_1 || {});
export var Namespace = Namespace_1;
```

*checks*
Our present emit is actually
```js
export var Foo;
(function (Foo) {
    Foo.cat = 2;
})(Foo || (Foo = {}));
export var Foo;
(function (Foo) {
    Foo.dog = 2;
})(Foo || (Foo = {}));
```
Which is in this case incorrect (it introduces duplicated exported identifiers) - I'll need to change the emit.